src/menu/stateful-container.js

```diff
@@ -195,21 +217,10 @@ export default class MenuStatefulContainer extends React.Component<
       ref: itemRef,
       isFocused: this.state.isFocused,
       isHighlighted: this.state.highlightedIndex === index,
-      onClick: (event: SyntheticMouseEvent<HTMLElement>) => {
-        if (this.props.onItemSelect && !item.disabled) {
-          this.props.onItemSelect({item, event});
-          this.internalSetState(STATE_CHANGE_TYPES.click, {
-            highlightedIndex: index,
-            activedescendantId,
-          });
-        }
-      },
-      onMouseEnter: () => {
-        this.internalSetState(STATE_CHANGE_TYPES.mouseEnter, {
-          highlightedIndex: index,
-          activedescendantId,
-        });
-      },
+      // binds so that in-line functions can be avoided. this is to ensure
+      // referential equality when option-list compares props in memoized compoent
+      onClick: this.handleItemClick.bind(this, activedescendantId, index, item),
+      onMouseEnter: this.handleMouseEnter.bind(this, activedescendantId, index),
```

Comment:
I think this will still create a new bound function every time the `getRequiredItemProps` is called.

Reply:
it maintains referential equality so that we can compare in memo function